src/dds_manager.cpp

```diff
@@ -929,21 +930,29 @@ bool DDSManager::replaceFilter(const std::string& topicName,
     decltype(m_sharedLock) lock(m_topicMutex);
     std::shared_ptr<TopicGroup> topicGroup = m_topics[topicName];
 
-    if (topicGroup->m_readerListeners.find(readerName) != topicGroup->m_readerListeners.end()) {
+    //NOTE: topicGroup->m_readerListeners.find(readerName) != topicGroup->m_readerListeners.end() was a cut and paste error from createSubscriber method
+    if (topicGroup->m_readerListeners.find(readerName) == topicGroup->m_readerListeners.end()) {
```

Comment:
```suggestion
    if (topicGroup->m_readerListeners.find(readerName) == topicGroup->m_readerListeners.end()) {
```
Got it - remove comment before merging

Reply:
That should have been removed.  Good find.